### PR TITLE
feat: added initial version of Frame Transaction types

### PR DESCRIPTION
**Pull request by Zizzamia**
**What changed? Why?**
Cooking some initial types for Frame transactions.

We are following the latest spec https://warpcast.notion.site/Frame-Transactions-Public-Draft-v2-9d9f9f4f527249519a41bd8d16165f73

We aim to enhance those types further as we gain insights from the latest spec version.

### CODE REVIEW

File: src/frame/types.ts [hunk 1]
```diff
@@ -115,29 +116,42 @@ export type FrameMetadataReact = FrameMetadataType & {
...
 /**
  * Note: exported as public Type
  */
 export type FrameMetadataResponse = Record<string, string>;
 
+/**
+ * Note: exported as public Type
+ */
+export type FrameTransactionResponse = {
+  chainId: `eip155:${number}`; // A CAIP-2 chain ID to identify the tx network
```
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/frame/types.ts:140]: Yeah you right, how would look like a type that supports Solana transactions as well?

[the remark this answers is not in the document]
Comment (cnasc) [reply src/frame/types.ts:140]: [The spec is pretty broad](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md#syntax) and TS [doesn't presently accommodate](https://github.com/microsoft/TypeScript/issues/41160) making a regex-validated string type.

Couple options:
```ts
// weakest
type ChainId = string;

// stronger, still weak
type ChainId = `${string}:${string}`;

// strongest, requires manual upkeep
type Namespace = 'eip155' | 'solana';
type Reference = string;
type ChainId = `${Namespace}:${Reference}`;
```

What might be best is to use a simpler type and just make sure we pass through some validation at runtime

File: src/frame/types.ts [hunk 2]
```diff
@@ -115,29 +116,42 @@ export type FrameMetadataReact = FrameMetadataType & {
...
+  params: FrameTransactionEthSendParams; // Specific parameters for chainId and method
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type FrameTransactionEthSendParams = {
+  abi: Abi; // The contract ABI for the contract to call.
+  data?: Hex; // The data to send with the transaction.
+  to: Hex; // The address of the contract to call.
```
[the remark this answers is not in the document]
Comment (robpolak) [reply src/frame/types.ts:151]: +1 Address seems to be more common with Viem

File: src/frame/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (snehkoul-cb) [reply src/frame/types.ts:140]: I worked on a package one year back [here](https://github.com/ChainAgnostic/caip-extended/tree/main) for chain agnostic which incorporates solana as well. We could use this lol

File: src/frame/types.ts [hunk 3]
```diff
@@ -115,29 +116,42 @@ export type FrameMetadataReact = FrameMetadataType & {
...
+  value: string; // The amount of Ether to send with the transaction.
```
[the remark this answers is not in the document]
Comment (robpolak) [reply src/frame/types.ts:152]: I would say both are okay here as long as we are very specific on formatting.  BigInt is more implicit to be WEI.

[the remark this answers is not in the document]
Comment (robpolak) [reply src/frame/types.ts:152]: Looking at the example, it does appear to be WEI : 

  params: {
    abi: [...], // JSON ABI of the function selector and any errors
    to: "0x00000000fcCe7f938e7aE6D3c335bD6a1a7c593D",
    data: ""
    value: "984316556204476",
  },

File: src/frame/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/frame/types.ts:151]: +1 making the change now

File: src/frame/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/frame/types.ts:152]: +1 making the change now

File: src/frame/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/frame/types.ts:140]: +1 making the change now

**Pull request merged**